horqrux/api.py

```diff
@@ -63,10 +63,24 @@ def ad_expectation(
     return inner(out_state, projected_state).real
 
 
+def ad_expectation(
+    state: Array, gates: GateSequence, observables: list[Primitive], values: dict[str, float]
+) -> Array:
+    """
+    Run 'state' through a sequence of 'gates' given parameters 'values'
+    and compute the expectation given an observable.
+    """
+    outputs = [
+        __ad_expectation_single_observable(state, gates, observable, values)
+        for observable in observables
+    ]
+    return jnp.stack(outputs)
+
+
 def expectation(
     state: Array,
     gates: GateSequence,
-    observable: GateSequence,
+    observable: list[Primitive],
```

Comment:
In some places you keep the initial notation, in other it is changed to observables. Observables is more readable, but observable was a list even before regardless of the number or single items.

Reply:
Good catch. I'll make this consistent.

The `list` type of `observable` before was confusing, since it never represented multiple observables in a list; it represented a composition of matrix multiplications.

Whereas now we really do take multiple observables, instead of a single composed observable, hence the rename. But the rename should be at least consistently applied, I agreed.